pkg/client/client.go

```diff
@@ -439,6 +439,31 @@ func (c *KpmClient) CompileTarPkg(tarPath string, opts *opt.CompileOptions) (*kc
 	return c.CompileWithOpts(opts)
 }
 
+// CompileGitPkg will compile the kcl package from the git url.
+func (c *KpmClient) CompileGitPkg(gitOpts *git.CloneOptions, compileOpts *opt.CompileOptions) (*kcl.KCLResultList, error) {
+	// 1. Create the temporary directory to pull the tar.
+	tmpDir, err := os.MkdirTemp("", "")
+	if err != nil {
+		return nil, reporter.NewErrorEvent(reporter.Bug, err, "internal bugs, please contact us to fix it.")
+	}
+	// clean the temp dir.
+	defer os.RemoveAll(tmpDir)
+
+	// 2. clone the git repo
+	_, err = git.CloneWithOpts(
```

Comment:
Fill in all the options of the CloneOptions.

```
_, err = git.CloneWithOpts(
                git.WithCommit(gitOpts.Commit)
                git.WithBranch(gitOpts.Branch)
                git.WithTag(gitOpts.Tag),
		git.WithRepoURL(gitOpts.RepoURL),
		git.WithLocalPath(tmpDir),
)
```

Reply:
Done!